### PR TITLE
Add Expires to PutObject opts

### PR DESCRIPTION
**Pull request by vadmeste**
Add Expires field while uploading an object

### CODE REVIEW

File: .github/workflows/vulncheck.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        go-version: [ 1.21.0 ]
+        go-version: [ 1.21.x ]
```
Comment (harshavardhana) [inline .github/workflows/vulncheck.yml:17]:
```suggestion
        go-version: [ 1.21.4 ]
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Vulncheck must be set to exact version of Go

**Review by harshavardhana — APPROVED** [review]

**Comment by MB175** [comment]
I tried using this option but its not working

**Comment by MB175** [comment]
@klauspost  I just did it looks like the header is set correctly also the "Expires" field is set. Looks like the issue lies within my minio server which is still serving the file after expiration strange enough.

So no worries your changes work as expected, sorry for the disturbance

**Comment by klauspost** [comment]
@MB175 The header has no functional meaning other than being stored on the object. It can be picked up by proxies, etc.

MinIO will do nothing more than store it and return whatever value it was set to. It is not a TTL. See [Object Lifecycle Management](https://min.io/docs/minio/linux/administration/object-management/object-lifecycle-management.html#object-expiration) on how to expire objects via lifecycle. This is similar to AWS S3.

**Pull request merged**